qldbdriver/qldbdriver.go

```diff
@@ -59,6 +59,13 @@ type semaphore struct {
 // Note that qldbSession.Client.Config.MaxRetries will be set to 0. This property should not be modified.
 // DriverOptions.RetryLimit is unrelated to this property, but should be used if it is desired to modify the amount of retires for statement executions.
 func New(ledgerName string, qldbSession *qldbsession.QLDBSession, fns ...func(*DriverOptions)) (*QLDBDriver, error) {
+	if qldbSession == nil {
+		return nil, &qldbDriverError{"Provided QLDBSession is nil."}
+	}
```

Comment:
I think this check is fine, but the bottom one is unnecessary.

Reply:
The reason for the 2nd check is to ensure that the dereference of `qldbSession.Client` on line 84 is safe.
Also the logic of setting `MaxRetries` is assuming a non-nil client.

Reply:
Now the dereference and `MaxRetries` logic only happens when we have a non-nil `qldbSession.Client`.

Reply:
Just curious how could customer pass in a nil client?

Reply:
@allimn It's a public constructor so we can't know for sure what might be passed in.